### PR TITLE
PICARD-2535: Strip trailing null chars for Vorbis / FLAC

### PR DESCRIPTION
**Pull request by phw**
# Summary

* This is a…
  * [x] Bug fix
  * [ ] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-2535, PICARD-2534

Remove trailing null chars in Vorbis / FLAC tags when loading and saving. This fixes specifically the reported issue with loading dates with trailing null char (PICARD-2534), but might also help in other cases.

# Solution
Remove trailing null char when loading and saving Vorbis comments.

I for now limited this to Vorbis, as all reported cases involved FLAC files. I was thinking about adding this to the `Metadata`  class in general, but was hesitant. For MP4 and ID3 we have some very specific values, e.g. the iTunes values, and I'm not fully sure we might break something here.

### CODE REVIEW

File: test/formats/test_vorbis.py [hunk 1]
```diff
@@ -189,6 +189,18 @@ def test_delete_invalid_tagname(self):
                 del metadata[invalid_tag]
                 save_metadata(self.filename, metadata)
 
+        @skipUnlessTestfile
+        def test_load_strip_trailing_null_char(self):
+            # Create a test file that contains only disctotal / tracktotal,
+            # but not totaldiscs and totaltracks
```
Comment (zas) [inline test/formats/test_vorbis.py:195]: This comment looks unrelated to this change, I guess it was copied from `test_delete_totaldiscs_totaltracks()`

**Review by zas — CHANGES_REQUESTED** [review]

File: test/formats/test_vorbis.py (hunk 1 above)
Comment (phw) [reply test/formats/test_vorbis.py]: Yes, this was a copy and paste left-over. Removed

**Review by zas — APPROVED** [review]
Lgtm
